Host/Domain change support added for local avatar url

jayedul opened the pull request:
<!--
Filling out this template is required.  Any PR that does not include enough information to be reviewed may be closed at a maintainers' discretion.  All new code requires documentation and tests to ensure against regressions.
-->

### Description of the Change
Current domain/host will be used for local avatar URLs even after site migration.

<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR. -->
Closes #64 

### How to test the Change

- Simply edit the image url host in `simple_local_avatar` user meta, maybe in database directly for easy test.
- Reload the page wherever the avatar image is supposed to load
- The image should be regenerated for that specific avatar size and the URL should get back to its original state based on current host/domain. 

### Changelog Entry
<!--
Please include a summary for this PR, noting whether this is something being Added / Changed / Deprecated / Removed / Fixed / or Security related.  You can replace the sample entries after this comment block with the single changelog entry line for this PR. -->
> Fixed - Local avatar urls remain old after domain/host change


### Credits
<!-- Please list any and all contributors on this PR so that they can be added to this projects CREDITS.md file. -->
Props @jayedul 


### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests pass.

jayedul committed:
Host/Domain change support added for local avatar url

jayedul committed:
PHPUnit Fix

github-actions[bot] committed:
Merge branch 'develop' into fix/64

github-actions[bot] committed:
Merge branch 'develop' into fix/64

ravinderk commented:
@jayedul Is this pull request ready for review?

github-actions[bot] committed:
Merge branch 'develop' into fix/64

jeffpaul commented:
@ravinderk let's go ahead and proceed with review on the PR

ravinderk commented on includes/class-simple-local-avatars.php:
```diff
@@ -371,12 +371,15 @@ public function get_simple_local_avatar_url( $id_or_email, $size ) {
 			if ( ! $avatar_full_path ) {
 				return '';
 			}
+
```
@jayedul  Can we make the following improvements to this function:

- Can we remove `! empty( $local_avatars['media_id'] )` check on line `357` because we already validate it last line `340`?
- We should generate a `full` image URL only if:
   - The `full` image URL does not exist
   - or The `full` image URL does not point to the same site upload directory

ravinderk reviewed (changes requested):
@jayedul I left one suggestion. Let me know if you have questions.

github-actions[bot] committed:
Merge branch 'develop' into fix/64

ravinderk committed:
chore: ignore file

ravinderk committed:
format: improve code formatting

ravinderk committed:
format: remove extra newline

ravinderk committed:
tests: set content url to example.com

ravinderk committed:
fix: return avatar if exist

ravinderk committed:
refactor: remove unused if condition

ravinderk committed:
refactor: remove unused if condition

jeffpaul commented:
@faisal-alvi can you help with the PHPUnit failures here as well as general testing/code review to help get this ready for merge/release?

faisal-alvi committed:
update unit test as per new changes

faisal-alvi committed:
improve condition to fix cypress fatal error

Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, string given in /var/www/html/wp-content/plugins/simple-local-avatars/includes/class-simple-local-avatars.php:342

faisal-alvi committed:
fixing phpcs issues

faisal-alvi commented:
The PHPUnit and Cypress test failures are fixed.

Also, I've tested this in a single setup site, and it's working perfectly. I'd like to move forward with testing it in a multisite network setup before we proceed with the merge. Hopefully, I'll have some time next week to conduct that testing.

faisal-alvi reviewed (approved):
It was tested in a multisite network, working as expected. Thanks for the work, everyone!

faisal-alvi commented:
Note: Unable to merge, requires one more approval. cc @ravinderk

ravinderk commented:
@faisal-alvi pr approved 🤞
